.github/workflows/rust.yml

```diff
@@ -30,7 +30,22 @@ jobs:
     if: ${{ github.event_name == 'push' || !github.event.pull_request.draft }}
     steps:
       - uses: actions/checkout@ee0669bd1cc54295c223e0bb666b733df41de1c5 # v2
-      - name: install rustfmt
+      - name: Install rustfmt
         run: rustup component add rustfmt
       - name: Check formatting
         run: cargo fmt -- --check
+
+  check-clippy:
+    needs: setup
+    runs-on: ubuntu-24.04
+    if: ${{ github.event_name == 'push' || !github.event.pull_request.draft }}
+    steps:
+      - uses: actions/checkout@ee0669bd1cc54295c223e0bb666b733df41de1c5 # v2
+      - name: Update repos
+        run: sudo apt update
+      - name: Install dependencies
+        run: sudo apt install -y libgtk-4-dev pkg-config libvulkan-dev libdrm-dev curl fuse3 libfuse-dev ocl-icd-libopencl1 ocl-icd-opencl-dev
+      - name: Install clippy
+        run: rustup component add clippy
+      - name: Run clippy
+        run: cargo clippy --verbose
```

Comment:
Could the check be done inside of the same job as the tests, in order to reuse the installed dependencies and build cache? Or is the intention to have it as a separate error from failed build/tests?

Reply:
@ilya-zlobintsev intention is to have separate error, but you decide

Reply:
The action is failing: https://github.com/ilya-zlobintsev/LACT/actions/runs/17332647025/workflow